pkg/controller/seed-controller-manager/default-application-controller/controller.go

```diff
@@ -0,0 +1,442 @@
+		return fmt.Errorf("failed to get usercluster client: %w", err)
+	}
+
+	// First ensure that the namespace exists
+	namespace := &corev1.Namespace{}
+	if err := userClusterClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: application.Namespace}, namespace); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get namespace: %w", err)
+		}
+
+		// Create the namespace if it doesn't exist
+		namespace.Name = application.Namespace
+		err := userClusterClient.Create(ctx, namespace)
+		if err != nil {
+			return fmt.Errorf("failed to create namespace: %w", err)
+		}
+	}
+
+	existingApplication := &appskubermaticv1.ApplicationInstallation{}
+	if err := userClusterClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: application.Name, Namespace: application.Namespace}, existingApplication); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get application installation: %w", err)
+		}
+
+		// Create the application
+		err := userClusterClient.Create(ctx, &application)
+		if err != nil {
+			return fmt.Errorf("failed to create application installation: %w", err)
+		}
+		return nil
+	}
+
+	// If the application is not enforced then we can't update it.
+	if application.Annotations[appskubermaticv1.ApplicationEnforceAnnotation] != AnnotationTrueValue {
+		return nil
+	}
+
+	// Before comparison delete the kubectl.kubernetes.io/last-applied-configuration annotation.
+	// This annotation is automatically generated by kubectl when applying a resource
+	// and causes unnecessary diffs
+	delete(existingApplication.Annotations, corev1.LastAppliedConfigAnnotation)
+
+	// Application installation already exists, update it if needed
+	if equality.Semantic.DeepEqual(existingApplication.Spec, application.Spec) &&
+		equality.Semantic.DeepEqual(existingApplication.Labels, application.Labels) &&
+		equality.Semantic.DeepEqual(existingApplication.Annotations, application.Annotations) {
+		return nil
+	}
+
+	// Required to update the object.
+	application.ResourceVersion = existingApplication.ResourceVersion
+	application.UID = existingApplication.UID
+
+	if err := userClusterClient.Update(ctx, &application); err != nil {
+		return fmt.Errorf("failed to update application installation: %w", err)
+	}
+	return nil
+}
+
+func (r *Reconciler) generateApplicationInstallation(application appskubermaticv1.ApplicationDefinition) appskubermaticv1.ApplicationInstallation {
+	appVersion := application.Spec.DefaultVersion
+	if appVersion == "" {
+		// Iterate through all the versions and find the latest one by semver comparison
+		for _, version := range application.Spec.Versions {
+			if appVersion == "" {
+				appVersion = version.Version
+			}
+			// Ensure both versions have the "v" prefix. This should never happen because our webhooks should prevent versions with `v` prefix from being created.
+			v1 := ensureVersionPrefix(version.Version)
+			v2 := ensureVersionPrefix(appVersion)
+
+			if semver.Compare(v1, v2) > 0 {
```

Comment:
Normally we use Masterminds/semver, which should have no problem parsing versions with or without the leading "v". Could then also use `.GreaterThan()` to improve readability even further.